### ISSUE TITLE
Runtime Exception when updating under 1.6.15

### ISSUE THREAD

**Issue by Romarain** [getgrav/grav#2670]
All is in the screenshot : 

![2019-09-18_13h30m57s_mintty](https://user-images.githubusercontent.com/22033256/65145502-05a1b780-da1a-11e9-8c06-d7009b7d297c.jpg)

**Comment by Romarain** [comment]
It broke my admin panel, but reinstalling the borken plugin manually (unzipping) fixed the problem. I would like to know wht this `folder.php` seems to have problems to run. Probably a permissions thing, but I've been through this topic the past few weeks, and added all the NTFS rights that I could on the project's root (all subfolders inherit from it). So I don't understand why the problem persists.

**Comment by rhukster** [comment]
Windows 10?

**Comment by Romarain** [comment]
Yes, Windows 10.

**Comment by rhukster** [comment]
Definitely a permission thing.  Do you have permission to delete the files? That's' where things seem to fail, delete is not successful.

**Comment by Romarain** [comment]
All permissions are allowed (reading, writing, executing). I'm under an admin account. UAC is OFF.

**Comment by rhukster** [comment]
We've looked at the code, and it seems solid.  It's used all the time in the admin plugin, and lots of people are using windows without this problem.  Must be something specific to your setup/system. Don't forget the webserver needs to have this permission so maybe caddy is running under a different user?

**Comment by Romarain** [comment]
I use the CLI, so maybe Git-bash is involved ? 
For Caddy, I run it under my account (Romain), not under the SYSTEM one, by using a CMD bat file.

**Comment by Romarain** [comment]
What's strange is that people manage to make it work under Windows, while I can't. I don't do anything more than using a batch file to run Caddy in order to make it use a caddyfile, and I use a Git-bash CLI to run Grav's commands.

**Comment by rhukster** [comment]
for a quick test commenting out the lines that are throwing the error (391-394 of Folder.php):

https://github.com/getgrav/grav/blob/develop/system/src/Grav/Common/Filesystem/Folder.php#L391-L394

**Comment by Romarain** [comment]
Wait : I've just thought about checking if I was launching the Git-bash CLI under administrative privilege, and I wasn't. So I added the option, and tested a `bin/gpm update` : I hadn't updated Quark theme yet, so I did it, and it passed. Maybe it's just luck (it's not a plugin), or maybe I fixed the problem.

**Comment by Romarain** [comment]
I'm gonna download an old version of a plugin, and manually install it, then I'll update it with the CLI to see if I'm right.

**Comment by Romarain** [comment]
If failed again.
So I commented the lines you suggested, and now, those ones are involved : 
![2019-09-18_14h49m45s_mintty](https://user-images.githubusercontent.com/22033256/65150096-a9dc2c00-da23-11e9-9788-76928b5287dc.jpg)

```
        if (file_exists($target)) {
            // Rename fails if target folder exists.
            throw new \RuntimeException('Cannot move files to existing folder/file.');
        }
```

**Comment by rhukster** [comment]
So your saying it deletes all files, but not the hidden `.git` folder? if so that's probably the reason the folder delete fails, as the folder is not empty.

**Comment by Romarain** [comment]
@rhukster Yeah. But am I supposed to know if that's the normal behavior of Grav ? Did I miss something during CLI updating ?

**Comment by rhukster** [comment]
It really should delete the hidden files along with the others.  It seems to work fine in all other circumstances.  maybe `scandir` on windows, or at least your instance doesn't include hidden dot files?

https://github.com/getgrav/grav/blob/develop/system/src/Grav/Common/Filesystem/Folder.php#L489

**Comment by Romarain** [comment]
After a few researches, I'm proud to say that I don't understand what to do, neither what I could have done so my instance (of what ?) doesn't include (?) hidden files. I've just installed Windows 10 recently, and reopened my Grav's files. Except the installation of Windows and a few softwares like git-bash (mine is portable) or 7zip, PHP, Caddy Server, I don't know what I could have done.

I've forced all executables (PHP, Caddy, Git) to work under administrative privilege, but this doesn't change anything. I think it is more something everybody _DO_ and that I ignore, than the opposite. I don't think that many people use the CLI under Windows, so we can say that "all is fine for most of people" ; moreover, to use Gravs command in Windows, we need to use a Linux-oriented CLI (Git-bash) ; but some Linux operations (deletion) could fail in Windows environment, who knows ? So maybe this explain why it doesn't work ? 

**I'm pretty sure it is the fault of Grav and not mine**, since it was perfectly working with the old versions (1.4, 1.5) and starts failing in 1.6.
